### PR TITLE
DataTypes challenge Of 30 Days Of JS

### PR DESCRIPTION
**Pull request by eloramirezcuadros**
Second day of JavaScript challenge🌷

### CODE REVIEW

**Review by anyruizd — COMMENTED** [review]
The exercises in this PR should be uncommented or in separate files.

File: day2OfJS/day2.js [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+
+/* 
+4. Calculate the person's total annual income by extracting the figures from the following text. 
+'He earns 5000 euros of salary per month, 10000 euros of annual bonus, 15000 euros of online courses per month'.
+let sum = 'He earns 5000 euros of salary per month, 10000 euros of annual bonus, 15000 euros of online courses per month';
+let matches = sum.match(/(\d+)/g)
+let mult1 = Number(matches[0]) * 12
+let mult2 = Number(matches[1])
+let mult3 = Number(matches[2]) * 12
```
Comment (josepplloo) [inline day2OfJS/day2.js]: The name of the variables need to have a name that helps to understand whats happening in the code, so:
eg.
`sum`: I expect a sum but I get a text;
`matches`: is too generic
`mult#`:  It do not have much context about the excercise

**Review by josepplloo — COMMENTED** [review]
In order to solve the exercise consider putting the answers(the console.log output), otherwise, you have to execute the code to know whats happen 
eg
```
\\15. Use indexOf to determine the position of the first occurrence of a in 30 Days Of JavaScript
const ocurr = '30 Days Of JavaScript';
console.log(ocurr.indexOf('a'));
\\ 4
 ```

File: day2OfJS/day2.js [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+
+//8. Check if the string contains a word Script using includes() method
+const check = 'JavaScript';
+console.log(check.includes('Script'));  //true
+
+
+//9. Split the string into an array using split() method
+const arr = 'JavaScript';
+console.log(arr.split(' '));  //['JavaScript']
```
Comment (josepplloo) [inline day2OfJS/day2.js]: instead `myString.split(' ')` use `myString.split('')` you will have  `\\ ["J", "a", "v", "a", "S", "c", "r", "i", "p", "t"]`

File: day2OfJS/day2.js [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+console.log(cut.substr(0, 4));  //Java
+
+
+//7. Slice out the phrase Days Of JavaScript from 30 Days Of JavaScript.
+const phrase = '30 Days Of JavaScript';
+console.log(phrase.substr(3, 21));  //Days Of JavaScript
+
+
+//8. Check if the string contains a word Script using includes() method
+const check = 'JavaScript';
```
Comment (josepplloo) [inline day2OfJS/day2.js]: look for a better name; instead `check` use `myWord` or `phrase` this comment applies for all variable's name, 🎀please take a moment to double-check🎀

File: day2OfJS/day2.js [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+console.log(check.includes('Script'));  //true
+
+
+//9. Split the string into an array using split() method
+const arr = 'JavaScript';
+console.log(arr.split(' '));  //['JavaScript']
+
+
+//10. Split the string 30 Days Of JavaScript at the space using split() method
+const arra = '30 Days Of JavaScript';
```
Comment (josepplloo) [inline day2OfJS/day2.js]: idem variable's name 👁️

File: day2OfJS/day2.js [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+
+//22. Use endsWith() method with the string 30 Days Of JavaScript and make the result true
+const strin_g2 = '30 Days Of JavaScript';
+console.log(strin_g2.endsWith('Script'));  //true
+
+
+//23. Use match() method to find all the a’s in 30 Days Of JavaScript
+const strin_g3 = '30 Days Of JavaScript';
+console.log(strin_g3.match(/of/gi));  //[ 'Of' ]
```
Comment (josepplloo) [inline day2OfJS/day2.js]: you look for `a` not for `of`

**Review by josepplloo — COMMENTED** [review]
I'm pausing the review, please double-check the exercises some are not complete, try to fix the variable's names

File: day2OfJS/day2.js [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+const challenge = '30 Days Of JavaScript';
+console.log(challenge); //30 Days Of JavaScript
+
+
+//2. Print the string on the browser console using console.log()
+console.log('Hi there, my name is Luisa'); //Hi there, my name is Luisa
+
+
+//3. Print the length of the string on the browser console using console.log()
+const name = 'JavaScript';
```
Comment (susird) [inline day2OfJS/day2.js]: Maybe a better fit of this variable could be `stringLength`

File: day2OfJS/day2.js [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+console.log(name.length); //9
+
+
+//4. Change all the string characters to capital letters using toUpperCase() method
+const mayus = 'JavaScript';
```
Comment (susird) [inline day2OfJS/day2.js]: A better fit for this variable name could be `stringToUpperCase`, remember to name the variables with a meaning of what are you doing

File: day2OfJS/day2.js [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+console.log(mayus.toUpperCase()); //JAVASCRIPT
+
+
+//5. Change all the string characters to lowercase letters using toLowerCase() method
+const minus = 'JavaScript';
```
Comment (susird) [inline day2OfJS/day2.js]: same here and the rest of the variables 😸

File: day2OfJS/day2.js [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+console.log(minus.toLowerCase()); //javascript
+
+
+//6. Cut (slice) out the first word of the string using substr() or substring() method
+const cut = 'JavaScript';
```
Comment (susird) [inline day2OfJS/day2.js]: Until here, all the exercises should be solved with the variable `const challenge = '30 Days Of JavaScript';` be sure to go over it and do the correction in the exercises

File: day2OfJS/day2.js [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+console.log(cut.substr(0, 4)); //Java
+
+
+//7. Slice out the phrase Days Of JavaScript from 30 Days Of JavaScript.
+const phrase = '30 Days Of JavaScript';
+console.log(phrase.substr(3, 21)); //Days Of JavaScript
+
+
+//8. Check if the string contains a word Script using includes() method
+const myWord = 'JavaScript';
```
Comment (susird) [inline day2OfJS/day2.js]: Same here `const challenge = '30 Days Of JavaScript';`

File: day2OfJS/day2.js [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+console.log(myWord.includes('Script')); //true
+
+
+//9. Split the string into an array using split() method
+const myString = 'JavaScript';
```
Comment (susird) [inline day2OfJS/day2.js]: same here `const challenge = '30 Days Of JavaScript';`

File: day2OfJS/day2.js [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+
+
+
+
+/*
+Exercise: Level 2
+1. Using console.log() print out the following statement:
+The quote 'There is no exercise better for the heart than reaching down and lifting people up.' by John Holmes teaches us to help one another.
+*/
+console.log("The quote 'There is no exercise better for the heart than reaching down and lifting people up.' by John Holmes teaches us to help one another.");
```
Comment (susird) [inline day2OfJS/day2.js]: Please, avoid doing all the logic inside the console.log statement, instead assign the result of the logic to a variable and console.log the variable.

File: day2OfJS/day2.js [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+//The quote 'There is no exercise better for the heart than reaching down and lifting people up.' by John Holmes teaches us to help one another.
+
+
+//2. Using console.log() print out the following quote by Mother Teresa:
+//"Love is not patronizing and charity isn't about pity, it is about love. Charity and love are the same -- with charity you give love, so don't just give money but reach out your hand instead."
+console.log("Love is not patronizing and charity isn't about pity, it is about love. Charity and love are the same -- with charity you give love, so don't just give money but reach out your hand instead.");
```
Comment (susird) [inline day2OfJS/day2.js]: same here

File: day2OfJS/day2.js [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+//Love is not patronizing and charity isn't about pity, it is about love. Charity and love are the same -- with charity you give love, so don't just give money but reach out your hand instead.
+
+
+//3. Check if typeof '10' is exactly equal to 10. If not make it exactly equal.
+console.log('10' == '10'); //true
```
Comment (susird) [inline day2OfJS/day2.js]: same here

File: day2OfJS/day2.js [hunk 15]
```diff
@@ -0,0 +1,234 @@
...
+
+
+//4. Check if parseFloat('9.8') is equal to 10 if not make it exactly equal with 10.
+const numFl = parseFloat('9.81');
+console.log(numFl === 10); //false
```
Comment (susird) [inline day2OfJS/day2.js]: Same here and the rest of the console.log from now on 🍭

File: day2OfJS/day2.js [hunk 16]
```diff
@@ -0,0 +1,234 @@
...
+*/
+
+
+//4. Calculate the person's total annual income by extracting the figures from the following text. 
+//'He earns 5000 euros of salary per month, 10000 euros of annual bonus, 15000 euros of online courses per month'.
+const salary = 'He earns 5000 euros of salary per month, 10000 euros of annual bonus, 15000 euros of online courses per month';
+const money = salary.match(/(\d+)/g);
+let multiplication1 = Number(money[0]) * 12;
+let multiplication2 = Number(money[1]);
+let multiplication3 = Number(money[2]) * 12;
```
Comment (susird) [inline day2OfJS/day2.js]: Remember to always use `const` unless you are going to change the value of the variable later on.

**Review by josepplloo — APPROVED** [review]
be careful with the names of the variables (always try to translate to double check the meaning) and with the non suggested use of regex

**Review by anyruizd — APPROVED** [review]
👏🏽 good job!

**Pull request merged**